src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -163,8 +165,72 @@ public boolean equals(Object o) {
         return true;
     }
 
+    /** return a MD5 sum for ths dictionary, the checksum is not re-computed each time this
+     * method is called. <pre>md5( (seq1.name+seq1.length) + ' '+(seq2.name+seq2.length) + ' '+...)</pre>
+     * @return a MD5 checksum for this dictionary or the empty string if it is empty
+     */
+    public String md5() {
+        if(isEmpty()) return "";
+        try {
+            MessageDigest md5 = MessageDigest.getInstance("MD5");
+            md5.reset();
+            for(SAMSequenceRecord ssr : mSequences)  {
+                if(ssr.getSequenceIndex()>0)  md5.update((byte)' ');
+                md5.update(ssr.getSequenceName().getBytes());
+                md5.update(String.valueOf(ssr.getSequenceLength()).getBytes());
+            }
+            String  hash = new BigInteger(1, md5.digest()).toString(16);
+            if (hash.length() != 32) {
+                final String zeros = "00000000000000000000000000000000";
+                hash = zeros.substring(0, 32 - hash.length()) + hash;
+            }
+            return hash;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
     @Override
     public int hashCode() {
         return mSequences.hashCode();
     }
+    
+    @Override
+    public String toString() {
+        return mSequences.toString();
+    }
+    
+    /** 
+     * @param sequenceIndex contig id
+     * @param position position in the contig
+     * @returns the number of bases seen from the first sequence at position 0 to the sequenceIndex/position or -1 on error*/
+    public long countBases(final int sequenceIndex,final int position)  {
+        if (sequenceIndex < 0 || position <0 || sequenceIndex >= mSequences.size() ) {
+            return -1L;
+        }
+        long countBases = 0L; /* number of bases seen so far */
+        for(int i = 0 ; i < sequenceIndex ; i++) countBases += getSequence(i).getSequenceLength();
+        
+        return countBases + position;
+    }
+    
+    /** shortcut to <pre>countBases(getSequenceIndex(sequenceName), position);</pre> */
+    public long  countBases(String sequenceName,final int position)  {
+        return countBases(getSequenceIndex(sequenceName), position);
+    }    
+    
+    /** 
```

Comment:
I would be hesitant to include these utility functions here as there hasn't been a great need for them and they are relatively easy to compute by client code.